### PR TITLE
vagrant: build and install vagrant-libvirt plugin

### PR DESCRIPTION
**Pull request by mat8913**
###### Motivation for this change

I was having trouble installing vagrant-libvirt with `vagrant plugin install vagrant-libvirt`, and I think it's cleaner to just let Nix manage everything.

#26202

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by globin** [comment]
@calbrecht could you have a look at this?

**Comment by calbrecht** [comment]
sure, i will

File: pkgs/development/tools/vagrant/0004-Support-system-installed-plugins.patch [hunk 1]
```diff
@@ -0,0 +1,96 @@
+From: Antonio Terceiro <terceiro@debian.org>
```
Comment (Mic92) [inline pkgs/development/tools/vagrant/0004-Support-system-installed-plugins.patch]: is this an upstream patch?

Comment (mat8913) [reply pkgs/development/tools/vagrant/0004-Support-system-installed-plugins.patch]: No, it's from Debian: https://sources.debian.org/src/vagrant/2.1.5+dfsg-1/debian/patches/0004-Support-system-installed-plugins.patch/

I made a couple of modifications to make it work with Nix.

File: pkgs/development/tools/vagrant/0004-Support-system-installed-plugins.patch [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+           FileUtils.mv(f.path, @path)
+         end
++      rescue Errno::EACCES
++        # Ignore permission denied against system-installed plugins; regular
++        # users are not supposed to write there.
++        raise unless @system
++      rescue Errno::EROFS
++        # Ignore read-only filesystem against system-installed plugins; regular
++        # users are not supposed to write there.
++        raise unless @system
```
[the remark this answers is not in the document]
Comment (mat8913) [reply pkgs/development/tools/vagrant/0004-Support-system-installed-plugins.patch]: I'm not sure.

File: pkgs/development/tools/vagrant/0004-Support-system-installed-plugins.patch (hunk 1 above)
Comment (mat8913) [reply pkgs/development/tools/vagrant/0004-Support-system-installed-plugins.patch]: Done

**Review by calbrecht — APPROVED** [review]
Tested installation in a user environment. ```vagrant-libvirt``` is present as system plugin afterwards and works as expected.

I'm just wondering why the sha256 changed for the ```mini_portile2``` gem.

**Comment by mat8913** [comment]
@calbrecht I think you might have accidentally put the wrong hash. `0gzfmcywp1da8nzfqsql2zqi648mfnx6qwkig3cv36n9m0yy676y` seems to be the hash for netrc, as seen here: https://github.com/NixOS/nixpkgs/blob/92a047a6c4d46a222e9c323ea85882d0a7a13af8/pkgs/servers/monitoring/sensu/gemset.nix#L198-L205

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**